pkg/sidecar/config_test.go

```diff
@@ -240,16 +279,20 @@ func TestValidateConfig(t *testing.T) {
 				SvidBundleFileName:    "bundle.pem",
 				RenewSignalDeprecated: "SIGHUP",
 			},
-			expectLogs: []string{"renewSignal will be deprecated, should be used as renew_signal"},
+			expectLogs: []shortEntry{{
+				Level:   logrus.WarnLevel,
+				Message: "renewSignal will be deprecated, should be used as renew_signal",
+			}},
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			log := &fakeLogger{}
+			log, hook := test.NewNullLogger()
 			tt.config.Log = log
-
 			err := ValidateConfig(tt.config)
 
-			require.Equal(t, tt.expectLogs, log.Warnings)
+			if tt.expectLogs != nil {
+				require.Equal(t, tt.expectLogs, getShortEntries(hook.AllEntries()))
```

Comment:
can you try to compare logs always? if no expected logs means that you must not get logs at all.